bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/internal/BlueGigaSerialHandler.java

```diff
@@ -198,14 +203,21 @@ public void close() {
     public void close(long timeout) {
         close = true;
         cancelTransactionTimer();
-        executor.shutdownNow();
         timer.cancel();
         try {
             parserThread.interrupt();
-            parserThread.join(timeout);
+            Thread.sleep(50);
```

Comment:
Using Thread.sleep is generally frowned upon but I think it would be justified in this case. Add a comment in the code so that other reviewers don't raise a fuss.